### PR TITLE
Make constants configurable via envars

### PR DESCRIPTION
**Pull request by Yadunund**
Addresses https://github.com/audrow/yatm/issues/375 by relying on environemnt variables instead of constants defined in src files.

- `YATM_TEST_CASE_CONFIG_PATH`: The absolute path to the YAML file that specifies the test case generation, as well as filters for what test cases to create from requirements and to specify the dimensions that should be varied for the test cases. See [example-test-case.config.yaml](./example-test-case.config.yaml)
- `YATM_REQUIREMENTS_DIRECTORY_PATH`: The absolute path to a directly containing YAML files files that describe testing requirements. See [Creating Requirements](#creating-requirements) for more details.
- `YATM_OUTPUT_DIRECTORY_PATH` (OPTIONAL): The absolute path to a directory where YATM should output test cases to. If unspecified, a `generated-files/` directory is created in the current working directory.
- `GITHUB_REPO_OWNER`: The name of the github organization or owner containing the repository below.
- `GITHUB_REPO_NAME`: The name of repository where Issue tickets for test cases should be opened.
- `GITHUB_TOKEN`: Your Github Personal Access token which has the ability to read and write to the repository above. 

Requirements that were stored here have been deleted and moved to `ros2_test_cases` via https://github.com/osrf/ros2_test_cases/pull/1057.

### CODE REVIEW

**Comment by Yadunund** [comment]
We should probably merge #368 in and rebase this PR to incorporate the core changes from that PR. (will need to move the DSITRO constant to an envar). In the meanwhile, @audrow what do you think of these changes?

File: src/constants.ts [hunk 1]
```diff
@@ -1,39 +1,65 @@
...
 
-export const CODE_URL = 'https://github.com/audrow/yatm'
+const packageXml = JSON.parse(
+  fs.readFileSync(join(process.cwd(), 'package.json'), 'utf8'),
+)
+export const VERSION = packageXml.version
+
+// =================================================================================================
+// Helper function to retrieve environment variables or throw and error if unspecified.
+export function extractStringEnvVar(
```
Comment (audrow) [inline src/constants.ts:52]: There's a really great library called "zod" that helps with this type of thing. Here's a post on how to use it to parse environment variables.

https://sergiodxa.com/articles/using-zod-to-safely-read-env-variables

File: README.md [hunk 2]
```diff
@@ -97,20 +98,16 @@ Here are the steps to using this package:
 
 ### Configuration
 
-At the moment things are not very well exposed or configurable without going into a few top-level files and configuring them. If this package has more interest, I expect it to become more configurable over time.
-Here is where different configuration lives:
+`YATM` relies on several environment variables that need to be set.
```
Comment (audrow) [inline README.md:101]: It might be worth mentioning direnv as a nice way to set this up.

https://direnv.net/

**Review by audrow — COMMENTED** [review]
Looks like a great change! I added a few comments, but looks good overall.

File: README.md (hunk 2 above)
Comment (Yadunund) [reply README.md:101]: [ff6b187](https://github.com/audrow/yatm/pull/404/commits/ff6b187cf123180d7f986e686a1e7c593c386098)

File: src/constants.ts (hunk 1 above)
Comment (Yadunund) [reply src/constants.ts]: Thanks for the suggestion https://github.com/audrow/yatm/pull/404/commits/45c6ed312d26e7fc966c4d8943fbd6ecf761c641

**Review by audrow — APPROVED** [review]
LGTM!

**Comment by audrow** [comment]
@Yadunund, I hope this was done. I didn't remember that I had mergify automerging PRs with approvals. 

I made a PR removing this #408 and merged it in.

You may want to rebase your PRs. You can use [`@mergifyio rebase`](https://docs.mergify.com/actions/rebase/), I believe.

**Pull request merged**